controllers/azuremanagedcontrolplane_controller_test.go

```diff
@@ -176,3 +183,80 @@ func TestAzureManagedControlPlaneReconcilePaused(t *testing.T) {
 	g.Expect(err).To(BeNil())
 	g.Expect(result.RequeueAfter).To(BeZero())
 }
+
+func TestAzureManagedControlPlaneReconcileNormal(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	cp := &infrav1.AzureManagedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-azmp",
+			Namespace: "fake-ns",
+		},
+		Spec:   infrav1.AzureManagedControlPlaneSpec{},
+		Status: infrav1.AzureManagedControlPlaneStatus{},
+	}
+	scheme, err := newScheme()
+	g.Expect(err).ToNot(HaveOccurred())
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(cp).WithStatusSubresource(cp).Build()
+	amcpr := &AzureManagedControlPlaneReconciler{
+		Client:   client,
+		Recorder: record.NewFakeRecorder(42),
+	}
+
+	helper, err := patch.NewHelper(cp, client)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	scopes := &scope.ManagedControlPlaneScope{
+		Cluster: &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "fake-cluster",
+				Namespace: "fake-ns",
+			},
+		},
+		Client:       client,
+		PatchHelper:  helper,
+		ControlPlane: cp,
+	}
+	scopes.SetAdminKubeconfigData(createFakeKubeConfig())
+	scopes.SetUserKubeconfigData(createFakeKubeConfig())
+	getNewAzureManagedControlPlaneReconciler = func(scope *scope.ManagedControlPlaneScope) (*azureManagedControlPlaneService, error) {
+		ctrlr := gomock.NewController(t)
+		defer ctrlr.Finish()
+		privateEndpointsClient := mock_privateendpoints.NewMockPrivateEndpointScope(ctrlr)
+		rGetter := []azure.ResourceSpecGetter{}
+		privateEndpointsClient.EXPECT().PrivateEndpointSpecs().Return(rGetter).AnyTimes()
+
+		peService := &privateendpoints.Service{
+			Scope:      privateEndpointsClient,
+			Reconciler: mock_async.NewMockReconciler(ctrlr),
+		}
+		return &azureManagedControlPlaneService{
+			kubeclient: scopes.Client,
+			scope:      scopes,
+			services: []azure.ServiceReconciler{
+				peService,
```

Comment:
Could we use a `*mock_azure.MockServiceReconciler` here instead of a concrete implementation of a particular service?

Reply:
@nojnhuh I believe if we do that, it will not walk the code path being tested because the mock will `EXPECT()` and `Return` whatever we like to obfuscate the functionality.  

We can use that and still provide coverage but it will lack the depth because that abstraction doesn't push through the intention of that test.  

If it's easier to do that, I can change this to match the suggestion with the understanding that this abstraction will not provide the confidence around changes to the functionality around services needing to "Reconcile" in it's own right. 

However we can move forward, I'm okay with.

Reply:
Let me know which way you'd rather me go with and I can make the changes @nojnhuh

Reply:
Which particular code path do you mean would no longer be exercised? Using a mock would skip over calling the real privateendpoints `Reconcile()`, but since that's already tested in isolation, I think it's acceptable to hide those details at this level of abstraction for the sake of testing `reconcileNormal()`.

Reply:
Perfect, I'll make the necessary changes and apply the squashing.

Reply:
This still fails because of the `reconcileKubeConfig` method at the end of `StoreClusterInfo` but removes the addition of the added serviceReconciler, etc.  

This should be good to go. Let me know what you think @nojnhuh